base/ca/src/main/java/com/netscape/cms/profile/constraint/P12ExportPasswordConstraint.java

```
@@ -0,0 +1,178 @@
+                String seq = password.substring(i, i + seqSize);
+                String invSeq = new StringBuilder(seq).reverse().toString();
+                if(password.indexOf(seq, i + seqSize) > 0 || password.indexOf(invSeq, i + seqSize) > 0){
+                    throw new ERejectException(CMS.getUserMessage(getLocale(req),
+                            "CMS_PROFILE_P12EXPORT_PASSWORD_SEQUENCE"));
+                }
+            }
+        }
+
+        param = getConfig(CONFIG_PASSWORD_MAX_REPEATED_CHAR);
+        if (!param.equals("")) {
+            int maxRepeat = Integer.parseInt(param);
+            int mostRepeatedChar = password.chars().mapToObj(x -> (char) x)
+                    .collect(Collectors.groupingBy(x -> x, Collectors.counting()))
+                    .entrySet().stream().max(Map.Entry.comparingByValue())
+                    .get().getValue().intValue();
+
+            if (maxRepeat < mostRepeatedChar) {
+                throw new ERejectException(CMS.getUserMessage(getLocale(req),
+                        "CMS_PROFILE_P12EXPORT_PASSWORD_REPEATED_CHAR", param));
+            }
+        }
+
+        if (getConfigBoolean(CONFIG_PASSWORD_CRACKLIB_CHECK)) {
+            try {
+                Process crack = new ProcessBuilder("/usr/sbin/cracklib-check").start();
+                BufferedWriter crackIn = crack.outputWriter();
+                BufferedReader crackOut = crack.inputReader();
+                crackIn.write(password);
+                crackIn.close();
+
+                String crackResult = crackOut.readLine().substring(password.length() + 2);
+                if (!crackResult.equals("OK")) {
+                    throw new ERejectException(CMS.getUserMessage(getLocale(req),
+                            "CMS_PROFILE_P12EXPORT_PASSWORD_CRACKLIB_FAILS", crackResult));
+                }
+
+            } catch (IOException e) {
+                logger.error(method + "impossible check password with cracklib.", e);
```

If cracklib check is enabled and it's failing I think we should throw an exception here.